### PR TITLE
Kotlin: Make the enum test more precise

### PR DESCRIPTION
**Pull request by igfoo**
In Kotlin 2 mode, there are more library classes matching `Enum%`, so the output was spuriously different.

### CODE REVIEW

File: java/ql/test/kotlin/library-tests/enum/test.ql [hunk 1]
```diff
@@ -1,7 +1,9 @@
 import java
 
-from Method m
-where m.getDeclaringType().getName().matches("Enum%")
-select m.getName()
+from Method m, Type t
+where
+  t = m.getDeclaringType() and
+  t.getName() = ["Enum<E>", "EnumUserKt"]
```
Comment (tamasvajk) [inline java/ql/test/kotlin/library-tests/enum/test.ql:6]: What are the other `Enum%` classes? Shouldn't those be included too?

**Review by tamasvajk — DISMISSED** [review]

File: java/ql/test/kotlin/library-tests/enum/test.ql (hunk 1 above)
Comment (igfoo) [reply java/ql/test/kotlin/library-tests/enum/test.ql:6]: Here's the results of
```
import java

from RefType t
where t.getName().matches("Enum%")
select t.getQualifiedName(), t.getName(), strictconcat(Method m | m.getDeclaringType() = t | m.getName(), ", ")
```
with 1.9.10. I thought that those 2 where the important ones, but I can add others if you think they would be useful?
```
| EnumUserKt | EnumUserKt | usesEnum |
| java.lang.Enum | Enum | clone, compareTo, describeConstable, equals, finalize, getDeclaringClass, hashCode, name, ordinal, toString, valueOf |
| java.lang.Enum$EnumDesc | EnumDesc | of, resolveConstantDesc, toString |
| java.lang.Enum<?> | Enum<?> | clone, compareTo, describeConstable, equals, finalize, getDeclaringClass, hashCode, name, ordinal, toString, valueOf |
| java.lang.Enum<AccessMode> | Enum<AccessMode> | clone, compareTo, describeConstable, equals, finalize, getDeclaringClass, hashCode, name, ordinal, toString, valueOf |
| java.lang.Enum<AccessType> | Enum<AccessType> | clone, compareTo, describeConstable, equals, finalize, getDeclaringClass, hashCode, name, ordinal, toString, valueOf |
| java.lang.Enum<AnnotationRetention> | Enum<AnnotationRetention> | clone, compareTo, describeConstable, equals, finalize, getDeclaringClass, hashCode, name, ordinal, toString, valueOf |
| java.lang.Enum<AnnotationTarget> | Enum<AnnotationTarget> | clone, compareTo, describeConstable, equals, finalize, getDeclaringClass, hashCode, name, ordinal, toString, valueOf |
| java.lang.Enum<ArrayAccess> | Enum<ArrayAccess> | clone, compareTo, describeConstable, equals, finalize, getDeclaringClass, hashCode, name, ordinal, toString, valueOf |
| java.lang.Enum<BasicType> | Enum<BasicType> | clone, compareTo, describeConstable, equals, finalize, getDeclaringClass, hashCode, name, ordinal, toString, valueOf |
| java.lang.Enum<Category> | Enum<Category> | clone, compareTo, describeConstable, equals, finalize, getDeclaringClass, hashCode, name, ordinal, toString, valueOf |
| java.lang.Enum<Characteristics> | Enum<Characteristics> | clone, compareTo, describeConstable, equals, finalize, getDeclaringClass, hashCode, name, ordinal, toString, valueOf |
| java.lang.Enum<ChronoField> | Enum<ChronoField> | clone, compareTo, describeConstable, equals, finalize, getDeclaringClass, hashCode, name, ordinal, toString, valueOf |
| java.lang.Enum<ChronoUnit> | Enum<ChronoUnit> | clone, compareTo, describeConstable, equals, finalize, getDeclaringClass, hashCode, name, ordinal, toString, valueOf |
| java.lang.Enum<ClassOption> | Enum<ClassOption> | clone, compareTo, describeConstable, equals, finalize, getDeclaringClass, hashCode, name, ordinal, toString, valueOf |
| java.lang.Enum<DayOfWeek> | Enum<DayOfWeek> | clone, compareTo, describeConstable, equals, finalize, getDeclaringClass, hashCode, name, ordinal, toString, valueOf |
| java.lang.Enum<DeprecationLevel> | Enum<DeprecationLevel> | clone, compareTo, describeConstable, equals, finalize, getDeclaringClass, hashCode, name, ordinal, toString, valueOf |
| java.lang.Enum<E> | Enum<E> | clone, compareTo, describeConstable, equals, finalize, getDeclaringClass, hashCode, name, ordinal, toString, valueOf |
| java.lang.Enum<ElementType> | Enum<ElementType> | clone, compareTo, describeConstable, equals, finalize, getDeclaringClass, hashCode, name, ordinal, toString, valueOf |
| java.lang.Enum<ExtendedOption> | Enum<ExtendedOption> | clone, compareTo, describeConstable, equals, finalize, getDeclaringClass, hashCode, name, ordinal, toString, valueOf |
| java.lang.Enum<FilteringMode> | Enum<FilteringMode> | clone, compareTo, describeConstable, equals, finalize, getDeclaringClass, hashCode, name, ordinal, toString, valueOf |
| java.lang.Enum<FormatStyle> | Enum<FormatStyle> | clone, compareTo, describeConstable, equals, finalize, getDeclaringClass, hashCode, name, ordinal, toString, valueOf |
| java.lang.Enum<Intrinsic> | Enum<Intrinsic> | clone, compareTo, describeConstable, equals, finalize, getDeclaringClass, hashCode, name, ordinal, toString, valueOf |
| java.lang.Enum<IsoCountryCode> | Enum<IsoCountryCode> | clone, compareTo, describeConstable, equals, finalize, getDeclaringClass, hashCode, name, ordinal, toString, valueOf |
| java.lang.Enum<IsoEra> | Enum<IsoEra> | clone, compareTo, describeConstable, equals, finalize, getDeclaringClass, hashCode, name, ordinal, toString, valueOf |
| java.lang.Enum<Kind> | Enum<Kind> | clone, compareTo, describeConstable, equals, finalize, getDeclaringClass, hashCode, name, ordinal, toString, valueOf |
| java.lang.Enum<Level> | Enum<Level> | clone, compareTo, describeConstable, equals, finalize, getDeclaringClass, hashCode, name, ordinal, toString, valueOf |
| java.lang.Enum<LinkOption> | Enum<LinkOption> | clone, compareTo, describeConstable, equals, finalize, getDeclaringClass, hashCode, name, ordinal, toString, valueOf |
| java.lang.Enum<Modifier> | Enum<Modifier> | clone, compareTo, describeConstable, equals, finalize, getDeclaringClass, hashCode, name, ordinal, toString, valueOf |
| java.lang.Enum<Month> | Enum<Month> | clone, compareTo, describeConstable, equals, finalize, getDeclaringClass, hashCode, name, ordinal, toString, valueOf |
| java.lang.Enum<Option> | Enum<Option> | clone, compareTo, describeConstable, equals, finalize, getDeclaringClass, hashCode, name, ordinal, toString, valueOf |
| java.lang.Enum<ResolverStyle> | Enum<ResolverStyle> | clone, compareTo, describeConstable, equals, finalize, getDeclaringClass, hashCode, name, ordinal, toString, valueOf |
| java.lang.Enum<RetentionPolicy> | Enum<RetentionPolicy> | clone, compareTo, describeConstable, equals, finalize, getDeclaringClass, hashCode, name, ordinal, toString, valueOf |
| java.lang.Enum<SettingsParser> | Enum<SettingsParser> | clone, compareTo, describeConstable, equals, finalize, getDeclaringClass, hashCode, name, ordinal, toString, valueOf |
| java.lang.Enum<SignStyle> | Enum<SignStyle> | clone, compareTo, describeConstable, equals, finalize, getDeclaringClass, hashCode, name, ordinal, toString, valueOf |
| java.lang.Enum<State> | Enum<State> | clone, compareTo, describeConstable, equals, finalize, getDeclaringClass, hashCode, name, ordinal, toString, valueOf |
| java.lang.Enum<Status> | Enum<Status> | clone, compareTo, describeConstable, equals, finalize, getDeclaringClass, hashCode, name, ordinal, toString, valueOf |
| java.lang.Enum<Tag> | Enum<Tag> | clone, compareTo, describeConstable, equals, finalize, getDeclaringClass, hashCode, name, ordinal, toString, valueOf |
| java.lang.Enum<TextStyle> | Enum<TextStyle> | clone, compareTo, describeConstable, equals, finalize, getDeclaringClass, hashCode, name, ordinal, toString, valueOf |
| java.lang.Enum<TimeDefinition> | Enum<TimeDefinition> | clone, compareTo, describeConstable, equals, finalize, getDeclaringClass, hashCode, name, ordinal, toString, valueOf |
| java.lang.Enum<TimeUnit> | Enum<TimeUnit> | clone, compareTo, describeConstable, equals, finalize, getDeclaringClass, hashCode, name, ordinal, toString, valueOf |
| java.lang.Enum<Type> | Enum<Type> | clone, compareTo, describeConstable, equals, finalize, getDeclaringClass, hashCode, name, ordinal, toString, valueOf |
| java.lang.Enum<UnicodeScript> | Enum<UnicodeScript> | clone, compareTo, describeConstable, equals, finalize, getDeclaringClass, hashCode, name, ordinal, toString, valueOf |
| java.lang.Enum<Wrapper> | Enum<Wrapper> | clone, compareTo, describeConstable, equals, finalize, getDeclaringClass, hashCode, name, ordinal, toString, valueOf |
| java.util.EnumSet | EnumSet | addAll, addRange, allOf, clone, complement, complementOf, copyOf, copyOf, noneOf, of, of, of, of, of, of, range, typeCheck, writeReplace |
| java.util.Enumeration | Enumeration | asIterator, hasMoreElements, nextElement |
| java.util.Enumeration<K> | Enumeration<K> | asIterator, hasMoreElements, nextElement |
| java.util.Enumeration<V> | Enumeration<V> | asIterator, hasMoreElements, nextElement |
| kotlin.Enum | Enum | clone, compareTo, describeConstable, equals, finalize, getDeclaringClass, hashCode, name, ordinal, toString |
| kotlin.enums.EnumEntries | EnumEntries | forEach, parallelStream, spliterator, stream, toArray |
```

Comment (tamasvajk) [reply java/ql/test/kotlin/library-tests/enum/test.ql:6]: Thanks for the list. 
Does it make sense to include `java.lang.Enum<E>`? That looks like a constructed generic type with a type argument `E`. Shouldn't we instead use `java.lang.Enum<?>` or `java.lang.Enum`? We could probably reduce the list to non-constructed types by adding a `t.getSourceDeclaration() = t` filter. `kotlin.Enum` also seems relevant.

Comment (igfoo) [reply java/ql/test/kotlin/library-tests/enum/test.ql:6]: `E` is the type from the test https://github.com/github/codeql/blob/main/java/ql/test/kotlin/library-tests/enum/enumUser.kt#L3 so I assume that we want that. I'll add `java.lang.Enum<?>`, `java.lang.Enum` and `kotlin.Enum`; we can always remove them later if they turn out to be more trouble than they're worth.

**Review by tamasvajk — APPROVED** [review]

**Pull request merged**